linera-storage-service/src/child.rs

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Zefchain Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+use linera_service::util::CommandExt;
+use std::time::Duration;
+use tokio::{
+    process::{Child, Command},
+    sync::{Semaphore, SemaphorePermit},
+};
+
+/// The storage service spanner
+pub struct StorageServiceChild {
+    endpoint: String,
+    binary: String,
+}
+
+/// The stores created by the `create_shared_test_store`
+/// are all pointing to the same storage.
+/// This is in contrast to other storage that are not
+/// persistent (e.g. memory) or uses a random table_name
+/// (e.g. RocksDb, DynamoDb, ScyllaDb)
+/// This requires two changes:
+/// * Only one test being run at a time with a semaphore.
+/// * After the test, the storage is cleaned
```

Comment:
Why don't we use a temporary subdirectory, so that multiple tests can run at the same time?

Reply:
Actually, from the point of view of the user whether the service is implementing memory or rocksDB is immaterial: he does not have to know. And anyway, memory is used.
But, what we can do is spawn memory services with different ports. That should work out reasonably well.